pkg/e2e/e2e.go

```diff
@@ -670,10 +670,18 @@ func cleanupAfterE2E(h *helper.H) (errors []error) {
 		log.Printf(msg, viper.GetString(config.Cluster.ID))
 
 		// Current default expiration is 6 hours.
-		// If the cluster hasn't been recycled, and the tests passed: Extend it 24h
-		if !viper.GetBool(config.Cluster.Reused) && clusterStatus != clusterproperties.StatusCompletedError {
-			if err := provider.ExtendExpiry(viper.GetString(config.Cluster.ID), 18, 0, 0); err != nil {
-				log.Printf("Error extending cluster expiration: %s", err.Error())
+		// If this cluster has addons, we don't want to extend the expiration
+		if !viper.GetBool(config.Cluster.Reused) && clusterStatus != clusterproperties.StatusCompletedError && viper.GetString(config.Addons.IDs) == "" {
+			if viper.GetString(config.Cluster.InstallSpecificNightly) != "" {
+				// For release-specific jobs, we want a shorter expiration for these: No more than 12h
+				if err := provider.ExtendExpiry(viper.GetString(config.Cluster.ID), 6, 0, 0); err != nil {
+					log.Printf("Error extending cluster expiration: %s", err.Error())
+				}
+			} else {
+				// If the cluster hasn't been recycled, and the tests passed: Extend it 24h
```

Comment:
If we get here, can't the cluster status be CompletedFailed? We only check for CompletedError above.

Reply:
lmfao kk